src/tests/curry.test.ts

```diff
@@ -151,15 +152,59 @@ describe('curry module', () => {
   })
 
   describe('debounce function', () => {
+    let func: DebounceFunction<any>
+    const mockFunc = jest.fn()
+    const runFunc3Times = () => _.iterate(3, func, undefined)
+
+    beforeEach(() => {
+      func = _.debounce({ delay: 600 }, mockFunc)
+
+    })
+    
+    afterEach(() => {
+      jest.clearAllMocks()
+    })
+    
     test('only executes once when called rapidly', async () => {
-      let calls = 0
-      const func = _.debounce({ delay: 600 }, () => calls++)
-      func()
+      runFunc3Times()
+      expect(mockFunc).toHaveBeenCalledTimes(0)
+      await _.sleep(610)
+      expect(mockFunc).toHaveBeenCalledTimes(1)
+    })
+
+    test("after calling cancel there should be no more debounce", () => { 
+      runFunc3Times()
+      expect(mockFunc).toHaveBeenCalledTimes(0)
+      func.cancel()
+      runFunc3Times()
+      expect(mockFunc).toHaveBeenCalledTimes(3)
+      runFunc3Times()
+      expect(mockFunc).toHaveBeenCalledTimes(6)
+    })
+
+    test("when we call the flush method it should execute the function immediately", () => {
+      func.flush()
+      expect(mockFunc).toHaveBeenCalledTimes(1)
+    })
+
+    test("I must ensure that when I call the flush method it will continue to debounce", async () => {
+      runFunc3Times()
+      expect(mockFunc).toHaveBeenCalledTimes(0)
+      func.flush()
+      expect(mockFunc).toHaveBeenCalledTimes(1)
       func()
+      expect(mockFunc).toHaveBeenCalledTimes(1)
+      await _.sleep(610)
```

Comment:
I would check #117 to see how @kmisha used the fake timers. I think this could be a flaky test because of the real timers/sleep.

Reply:
in this case we should wait to merge the other PR, because to use his suggestion we need jest to be updated to version 18

Reply:
If you need to update packages feel free to 👍 others can always rebase if this gets merged first.

Reply:
![image](https://user-images.githubusercontent.com/63638196/197867746-26e352e6-5012-4d13-87de-3db7e196e1d3.png)

in his pr jest was updated to version 18, can i update directly to 19.x?

Strange thing about libraries being outdated, I thought the bot updated them automatically

Reply:
Ya, the bot should be updating them but it's not. I'd love to update them all to their latest versions. Let me do that in a separate PR real quick, then you can rebase.

Reply:
I made a PR (#124) to update all the versions. As long as tests pass, I'll merge asap.

Reply:
Ok, it's been merged @MarlonPassos-git you can rebase off master now and get the upgraded jest package.